src/tools/rados/rados.cc

```diff
@@ -315,6 +317,50 @@ static int do_get(IoCtx& io_ctx, RadosStriper& striper,
   return ret;
 }
 
+static int do_repair_get(IoCtx& io_ctx, const char *objname,
+	const char *outfile, unsigned op_size, int32_t osdid, epoch_t epoch)
+{
+  string oid(objname);
+
+  int fd;
+  if (strcmp(outfile, "-") == 0) {
+    fd = STDOUT_FILENO;
+  } else {
+    fd = TEMP_FAILURE_RETRY(::open(outfile, O_WRONLY|O_CREAT|O_TRUNC, 0644));
+    if (fd < 0) {
+      int err = errno;
+      cerr << "failed to open file: " << cpp_strerror(err) << std::endl;
+      return -err;
+    }
+  }
+
+  uint64_t offset = 0;
+  int ret;
```

Comment:
could also initialize `ret` to `0`. and `break` instead of `goto out` if something goes wrong.